### PR TITLE
[TT-13155] Explicitly copy BaseMiddleware for each middleware that takes it

### PR DESCRIPTION
**Pull request by titpetric**
<details open>
  <summary><a href="https://tyktech.atlassian.net/browse/TT-13155" title="TT-13155" target="_blank">TT-13155</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
      <td>[Regression] Gateway Debug logs starting v5.3 only logs AccessRightsCheck for most of the middlewares</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Bug
        Bug
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Dev</td>
    </tr>
    <tr>
      <th>Points</th>
      <td>N/A</td>
    </tr>
    <tr>
      <th>Labels</th>
      <td><a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%20'24Bugsmash%20ORDER%20BY%20created%20DESC" title="'24Bugsmash">'24Bugsmash</a>, <a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%202025lts%20ORDER%20BY%20created%20DESC" title="2025lts">2025lts</a>, <a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%20SESAP%20ORDER%20BY%20created%20DESC" title="SESAP">SESAP</a>, <a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%20customer_bug%20ORDER%20BY%20created%20DESC" title="customer_bug">customer_bug</a>, <a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%20jira_escalated%20ORDER%20BY%20created%20DESC" title="jira_escalated">jira_escalated</a></td>
    </tr>
  </table>
</details>

---

### **PR Type**
Enhancement

PR:

- implements per-middleware basemiddleware copy behaviour
- reverts the logger+mutex on base middleware
- touches coprocess/ to better handle grpc server startup, shutdown, conflicts on static port number - not to swallow errors when net.Listener fails

___

### **Description**
- Refactored the `BaseMiddleware` initialization process by introducing a `NewBaseMiddleware` function, encapsulating the creation logic.
- Added a `Copy` method to `BaseMiddleware` to create scoped copies with a duplicated logger, ensuring middleware-specific logging.
- Updated all middleware initialization in `gateway/api_loader.go` to use `baseMid.Copy()` for better isolation and logging scope.
- Enhanced code readability and maintainability by centralizing `BaseMiddleware` creation logic and ensuring proper separation of concerns.

### CODE REVIEW

**Review by jeffy-mathew — APPROVED** [review]

File: .taskfiles/test.yml [hunk 1]
```diff
@@ -50,7 +50,8 @@ tasks:
       - rm -rf coverage && mkdir -p coverage
       - for: { var: packages, as: package }
         cmd: |-
-          gotestsum --no-color=false --hide-summary=skipped --raw-command \
+          gotestsum --jsonfile coverage/{{.product}}-{{.package | replace "." "gateway" | replace "/" "-"}}.json \
```
Comment (titpetric) [inline .taskfiles/test.yml:53]: This fixes `task test:report` for the target.

File: gateway/middleware.go [hunk 2]
```diff
@@ -48,8 +49,6 @@ type TykMiddleware interface {
 	GetSpec() *APISpec
 
 	Init()
-	SetName(string)
-	SetRequestLogger(*http.Request)
```
Comment (titpetric) [inline gateway/middleware.go:52]: Both are being removed from the interface as they are BaseMiddleware interface functions.

File: .taskfiles/test.yml [hunk 3]
```diff
@@ -86,8 +87,8 @@ tasks:
     vars:
       count: '{{ .count | default "10" }}'
     cmds:
-      - gotestsum --hide-summary=skipped --junitfile=unit-tests.xml --raw-command cat coverage/{{.product}}-all.json
-      - echo "Slowest {{.count}} tests:" && cat coverage/{{.product}}-all.json | gotestsum tool slowest | head -n {{.count}} | sed -e 's|{{.package}}/||g'
+      - gotestsum --hide-summary=skipped --junitfile=unit-tests.xml --raw-command cat coverage/*.json
+      - echo "Slowest {{.count}} tests:" && cat coverage/*.json | gotestsum tool slowest | head -n {{.count}} | sed -e 's|{{.package}}/||g'
```
Comment (titpetric) [inline .taskfiles/test.yml:92]: The above change makes `test:integration` and `test:report` (slowest tests) work together.

`cat coverage/*.json | gotestsum tool slowest`

By default gotestsum reports tests slower than 100ms, reports ~150 tests out of the set of ~2600.

File: coprocess/grpc/services_test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	)
+	coprocess.RegisterDispatcherServer(s, &dispatcher{})
+	return s
+}
+
+func startTestServices(tb testing.TB) (*gateway.Test, func()) {
+	tb.Helper()
+
+	listener, err := net.Listen("tcp", ":0")
+	require.NoError(tb, err)
```
Comment (titpetric) [inline coprocess/grpc/services_test.go:28]: Previously the listener would use the static port and never close the socket, no error check.
Anyway. It runs on a random port now.

**Pull request merged**